README.md

```diff
@@ -1,3 +1,101 @@
+# Harbor with Rackspace Managed Kubernetes Auth
+
+This section explains how to create the development environment for developing the integration code for Harbor with Rackspace Managed Kubernetes Auth on Mac OSX. This uses your Docker/Docker Compose only because this is a local dev env. Integration with Kubernetes happens in [kubernetes-installer](https://github.com/rcbops/kubernetes-installer).
+
+Read the [Harbor](#harbor) section. Because master is unstable, all development happens in the [rackspace-mk8s-auth branch](https://github.com/rcbops/kubernetes-harbor/tree/rackspace-mk8s-auth) which is always based on a stable release branch. The rackspace-mk8s-auth branch will need to be rebased onto newer Harbor stable branches as we upgrade to newer releases.
+
+The only thing in the master branch is this section of the README just so it's easy to find.
+
+## Dev Env
+
+There are a few things you need to do to get a local dev env on Mac OSX going.
+
+1. Fork the repo.
+
+1. Even though the repo is git@github.com:rcbops/kubernetes-harbor.git it's easier to clone it to a directory structure where Go lang expects everything to be.
+
+    ```bash
+    mkdir $GOPATH/src/github.com/vmware
+    git clone git@github.com:my-github-username/kubernetes-harbor.git $GOPATH/src/github.com/vmware/harbor
+    cd $GOPATH/src/github.com/vmware/harbor
+    git remote add upstream git@github.com:rcbops/kubernetes-harbor.git
+    ```
+
+1. Create a data dir.
+
+    ```bash
+    sudo mkdir /data
+    sudo chown $(whoami):staff /data
+    ```
+
+1. Open your Docker preferences and under File Sharing add dirs `/data` and `/var/log`.
+
+## Run and Configure Kubernetes Auth
+
+1. Fork and clone [kubernetes-auth](https://github.com/rcbops/kubernetes-auth).
+
+1. Run it with the dummy backend in the same network as Harbor.
+
+    ```bash
+    make run NETWORK_NAME_BASE=make_harbor
+    ```
+
+## Run Harbor
+
+This is cribbed from the [compile guide](docs/compile_guide.md). This deploys all Harbor components using your local Docker/Docker Compose. A few changes have been made to [harbor.cfg](make/harbor.cfg) to make this work out of the box.
+
+```bash
+make install GOBUILDIMAGE=golang:1.7.3 COMPILETAG=compile_golangimage CLARITYIMAGE=vmware/harbor-clarity-ui-builder:1.2.7
+```
+
+## Login to Harbor as admin
+
+1. Open [registry.127.0.0.1.nip.io](http://registry.127.0.0.1.nip.io).
+
+1. You can login using the username/password combo `admin/Harbor12345`. Becaue of the way Harbor is designed, the admin user is the one and only user that will use Harbor's own DB for authentication.
+
+## Login to Harbor as a user
+
+1. To login to Harbor as a user, you first need to create a user/token in the kubernetes-auth dummy backend.
+
+1. Work through the [kubernetes-auth example](https://github.com/rcbops/kubernetes-auth#example) to create a user.
```

Comment:
this guide asks people to run `make run` again, and to target `http://localhost:8080`. is that correct? won't containers on a user-created network have their own IP?

Reply:
even better, can we wrap all this in a bash script? it's pretty intimidating

Reply:
I've created a PR to remove that redundant `make run` https://github.com/rcbops/kubernetes-auth/pull/43

`http://localhost:8080` is correct. The containers do get their own IP but published ports are exposed on localhost through the magic of Docker for Mac.

I agree that it's pretty intimidating but to do development on this stuff you do need to understand it at that level of detail. A script would abstract that away and make understanding/modification for the developer more difficult down the line.